tests/func/plots/test_show.py

```diff
@@ -296,3 +296,20 @@ def test_collect_non_existing_dir(tmp_dir, dvc, run_copy_metrics):
     assert "error" in result["workspace"]["data"]["subdir"]
     # make sure others gets loaded
     assert result["workspace"]["data"]["plot.json"]["data"] == metric
+
+
+def test_config_output_dir(tmp_dir, dvc, capsys):
+    subdir = tmp_dir / "subdir"
+    ret = main(["config", "plots.out_dir", os.fspath(subdir)])
+    assert ret == 0
+
+    metric = [{"first_val": 100, "val": 2}, {"first_val": 200, "val": 3}]
+    (tmp_dir / "metric.json").dump_json(metric, sort_keys=True)
+
+    with tmp_dir.chdir():
+        assert main(["plots", "show", "metric.json"]) == 0
+
+    out, _ = capsys.readouterr()
+    assert subdir.as_uri() in out
+    assert subdir.is_dir()
+    assert (subdir / "index.html").is_file()
```

Comment:
Could you move it to `tests/integration/plots/test_json.py`? (the name is inadequate, but I will change it anyway in #7477).
Also, after the last call, could you ensure that providing `-o some_other_dir` overrides the config option?